File: app/src/main/java/com/example/lab6/MainActivity.java
package com.example.lab6;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

import java.io.InputStream;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {


    public static final String ERROR = "Exception";
    private EditText editText;
    private String addr;
    private class DownloadImageTask extends AsyncTask<String, Void, Bitmap> {
        ImageView imageView;

        public DownloadImageTask(ImageView imageView) {
            this.imageView = imageView;
        }

        protected Bitmap doInBackground(String... urls) {
            String urldisplay = urls[0];
            Bitmap image = null;
            try {
                InputStream input = new java.net.URL(urldisplay).openStream();
                image = BitmapFactory.decodeStream(input);
            } catch (Exception e) {
                Log.e(ERROR, e.getMessage());
            }
            return image;
        }
        protected void onPostExecute(Bitmap result) {
            if(result == null) imageView.setImageResource(android.R.color.transparent);
            else imageView.setImageBitmap(result);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.editText);
        findViewById(R.id.download).setOnClickListener(this);
        findViewById(R.id.clear).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.clear){editText.setText("");}
        else {
            addr = editText.getText().toString();
            new DownloadImageTask((ImageView) findViewById(R.id.imageView)).execute(addr);
        }
    }
}

File: task3/src/main/java/com/example/task3/MainActivity.kt
package com.example.task3

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.coroutineScope
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.*
import java.net.URL

class MainActivity : AppCompatActivity() {
    val ERROR = "Exception"
    var job: Job? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var addr: String
        var image: Bitmap? = null
        val imageView: ImageView = findViewById(R.id.imageView)
        val editText: EditText? = findViewById(R.id.editText)
        val download: Button = findViewById(R.id.download)
        val clear: Button = findViewById(R.id.clear)
        clear.setOnClickListener {editText!!.setText("")}
        download.setOnClickListener {
             lifecycle.coroutineScope.launchWhenResumed {
                withContext(Dispatchers.IO) {
                    try {
                        addr = editText?.text.toString()
                        val input = URL(addr).openStream()
                        image = BitmapFactory.decodeStream(input)
                    } catch (e: Exception) {
                        Log.e(ERROR, e.message!!)
                    }
                }
                if (image != null) imageView.setImageBitmap(image)
                else imageView.setImageResource(android.R.color.transparent)
            }
        }
    }

}

